utils/land-patterns.stanza

```diff
@@ -1676,6 +1676,46 @@ public defn two-pin-chip-landpattern (length:Toleranced,
     DENSITY-LEVEL, polarized?
   )
 
+public defn parse-chip-package-code (code:String) -> Dims : 
+  ; helper function to insert decimal point after first character.
+  defn insert-decimal (s:String) -> String :
+    to-string $ "%_.%_" % [s[0], s[1 to false]] 
+
+  ; validate the code :
+  ;   - must be all numeric (TBD: extract metric/imperial information before calling)
+  ;   - must be 4, 5, or 6 characters long
+  if find({not digit?(_)}, code) is-not False or 
```

Comment:
The first condition is a double negative, so that makes it hard to follow. The documentation above is great!
`not all?(digit?, code)`